### PR TITLE
Add profiling

### PR DESCRIPTION
**Pull request by jeremypoulter**
Added some basic profiling code to help identify code that is blocking the CPU and stopping other stuff happening.

### CODE REVIEW

File: src/emonesp.h [hunk 1]
```diff
@@ -5,15 +5,7 @@
 // General support code used by all modules
 // -------------------------------------------------------------------
 
-// Uncomment to use hardware UART 1 for debug else use UART 0
-// #define DEBUG_SERIAL1
```
Comment (sandeen) [inline src/emonesp.h]: Ok, good, this fixes that regression I mentioned elsewhere but I'd still liketo know what happened therek and what else may have been (inadvertently?) changed ..

Comment (jeremypoulter) [inline src/emonesp.h]: I think the default should be Serial1 for debug as Serial is connected to OpenEVSE. However I do use Serial for debug sometime when just developing without a connected OpenEVSE. Was probably a mistake to commit anything else. I have also been transitioning to the macros in debug.h as these go away in production builds (I think I will rename openevse-ota to openevse-dev)

Comment (sandeen) [inline src/emonesp.h]: Yep, it's just that the default build seems to be outputting debug on the main serial line, which is a bug IMHO.

Comment (glynhudson) [inline src/emonesp.h]: Yes, very sorry this was a mistake to commit this. Debug should not go to the main serial.

**Pull request merged**